src/drive/web/modules/views/Trash/index.jsx

```diff
@@ -0,0 +1,117 @@
+import File from '../FileWithActions'
+import RealTimeQueries from '../Drive/RealTimeQueries'
+
+import { buildTrashQuery } from 'drive/web/modules/queries'
+
+export const TrashView = ({ router, children }) => {
+  const { isBigThumbnail, toggleThumbnailSize } = useContext(
+    ThumbnailSizeContext
+  )
+
+  const trashQuery = buildTrashQuery()
+
+  const result = useQuery(trashQuery.definition, trashQuery.options)
+  const isInError = result.fetchStatus === 'error'
+  const hasDataToShow = !isInError && (result.data && result.data.length > 0)
+  const isLoading = !hasDataToShow && result.fetchStatus === 'loading'
+  const isEmpty = !isLoading && !hasDataToShow
+  const navigateToFolder = useCallback(folderId => {
+    router.push(`/trash/${folderId}`)
+  })
+
+  const navigateToFile = useCallback(file => {
+    router.push(`/trash/file/${file.id}`)
+  })
+
+  return (
+    <Main>
+      <RealTimeQueries doctype="io.cozy.files" />
+      <ModalStack />
+      <Topbar>
+        <Breadcrumb />
+        <Toolbar canUpload={false} canCreateFolder={false} disabled={false} />
+      </Topbar>
+
+      <SelectionBar />
+      <FileListv2>
+        <MobileFileListHeader
+          folderId={null}
+          canSort={false}
+          thumbnailSizeBig={isBigThumbnail}
+          toggleThumbnailSize={toggleThumbnailSize}
+        />
+        <FileListHeader
+          folderId={null}
+          canSort={false}
+          thumbnailSizeBig={isBigThumbnail}
+          toggleThumbnailSize={toggleThumbnailSize}
+        />
+        <FileListBodyV2 selectionModeActive={false}>
+          {isInError && <Oops />}
+          {isLoading && <FileListRowsPlaceholder />}
+          {isEmpty && <EmptyDrive canUpload={false} />}
+          {hasDataToShow && (
+            <div className={isMobileApp() ? 'u-ov-hidden' : ''}>
+              {result.data.map(file => (
+                <File
+                  key={file._id}
+                  attributes={file}
+                  displayedFolder={null}
+                  onFolderOpen={navigateToFolder}
+                  onFileOpen={navigateToFile}
+                  withSelectionCheckbox={true}
+                  withFilePath={false}
+                  withSharedBadge={true}
+                  isFlatDomain={true}
+                  thumbnailSizeBig={isBigThumbnail}
+                />
+              ))}
+              {result.hasMore && <LoadMore fetchMore={result.fetchMore} />}
+            </div>
+          )}
+        </FileListBodyV2>
+      </FileListv2>
+      {children}
+    </Main>
+  )
+}
+
+const TrashViewWithProvider = props => (
+  <SharingProvider doctype="io.cozy.files" documentType="Files">
+    <ThumbnailSizeContextProvider>
+      <RouterContextProvider>
+        <ModalContextProvider>
+          <TrashView {...props} />
+        </ModalContextProvider>
+      </RouterContextProvider>
+    </ThumbnailSizeContextProvider>
+  </SharingProvider>
+)
+
+export default TrashViewWithProvider
```

Comment:
Nit: As a reviewer, I expect this file to be refactored in the future, but for the sake of the reviewer, I think it'd be a bit better if the refactor was done before the new view was created, so that I do not have to guess that it's going to be refactored afterwards.